src/metax_api/api/rest/base/serializers/catalog_record_serializer.py

```diff
@@ -88,10 +88,23 @@ def create(self, validated_data):
 
     def to_representation(self, instance):
         res = super(CatalogRecordSerializer, self).to_representation(instance)
-        res['data_catalog'] = DataCatalogSerializer(instance.data_catalog).data
+
+        if self.expand_relation_requested('data_catalog'):
+            res['data_catalog'] = DataCatalogSerializer(instance.data_catalog).data
+        else:
+            res['data_catalog'] = {
+                'id': instance.data_catalog.id,
+                'identifier': instance.data_catalog.catalog_json['identifier'],
+            }
 
         if 'contract' in res:
-            res['contract'] = ContractSerializer(instance.contract).data
+            if self.expand_relation_requested('data_catalog'):
```

Comment:
Should this be if self.expand_relation_requested('contract'): ?

Reply:
Very much yes. The sole test case had both data_catalog and contract as parameters, so naturally Murphy sees an opportunity here...